### PR TITLE
add SubjectInformationAccess extension support

### LINKED ISSUE #5294: Req: add support for "Subject Information Access" X509v3 extension
WRT #1947:

> There's also a set of extensions we've chosen not to implement at all until users ask for them:
>
> [...]
>
> - [ ] Subject Information Access (same as AIA)
>
> [...]

I'd like to "officially" make a request support for the "Subject Information Access" extension described in [RFC 5280, section 4.2.2.2](https://tools.ietf.org/html/rfc5280#section-4.2.2.2):

>  The subject information access extension indicates how to access information and services for the subject of the certificate in which the extension appears. When the subject is a CA, information and services may include certificate validation services and CA policy data. When the subject is an end entity, the information describes the type of services offered and how to access them. In this case, the contents of this extension are defined in the protocol specifications for the supported services. This extension may be included in end entity or CA certificates. Conforming CAs MUST mark this extension as non-critical.

As mentioned in #1947 (and quoted above), SIA is nearly identical to AIA:

```
id-pe-authorityInfoAccess OBJECT IDENTIFIER ::= { id-pe 1 }

    AuthorityInfoAccessSyntax  ::=
        SEQUENCE SIZE (1..MAX) OF AccessDescription

    AccessDescription  ::=  SEQUENCE {
        accessMethod          OBJECT IDENTIFIER,
        accessLocation        GeneralName  }
```

```
id-pe-subjectInfoAccess OBJECT IDENTIFIER ::= { id-pe 11 }

    SubjectInfoAccessSyntax  ::=
        SEQUENCE SIZE (1..MAX) OF AccessDescription

    AccessDescription  ::=  SEQUENCE {
        accessMethod          OBJECT IDENTIFIER,
        accessLocation        GeneralName  }
```

In an attempt to narrow the scope of this request to fit my actual use case (since there have apparently been zero requests for support for the SIA extension in the last five years!), I am specifically interested in the ability to create a new X.509v3 certificate containing this extension (OID `1.3.6.1.5.5.7.1.11`) with (multiple instances of) the"`id-ad-caRepository`" (OID `1.3.6.1.5.5.7.48.5`) "`accessMethod`" pointing to different HTTP URIs ("`uniformResourceIdentifier`").

Thanks for your consideration!

### PR DESCRIPTION
**Pull request by reaperhulk**
fixes #5294

### CODE REVIEW

File: docs/x509/reference.rst [hunk 1]
```diff
@@ -2973,6 +3003,23 @@ instances. The following common OIDs are available as constants.
...
+
+    .. versionadded:: 3.0
+
+    .. attribute:: CA_REPOSITORY
+
+        Corresponds to the dotted string ``"1.3.6.1.5.5.7.48.5"``. Used as the
+        identifier for CA repository data in
+        :class:`~cryptography.x509.AccessDescription` objects.
+
+    .. attribute:: CA_ISSUERS
```
Comment (alex) [inline docs/x509/reference.rst:3016]: This doesn't seem to exist?

Comment (jlgaddis) [reply docs/x509/reference.rst:3016]: > This doesn't seem to exist?

Not as a valid `accessMethod` for SIA, anyways.

According to the syntax shown at the very end of section 4.2.2.2, the only valid OIDs for `accessMethod` (in SIA) are `1.3.6.1.5.5.7.48.5` ("`id-ad-caRepository`") and `1.3.6.1.5.5.7.48.3` ("`id-ad-timeStamping`"); the former is "used when the subject is a CA" and the latter is "used when the subject is an end entity".

`1.3.6.1.5.5.7.48.2` is "`id-ad-caIssuers`", though, so perhaps just a classic "off-by-one" error?  :smile:

---

On a (somewhat) related note, [section 3.1 of RFC4387](https://tools.ietf.org/html/rfc4387#section-3.1), identifies at least two other valid OIDs for `accessMethod` (in AIA/SIA):

* `1.3.6.1.5.5.7.48.6`: "`id-ad-http-certs`"
* `1.3.6.1.5.5.7.48.7`: "`id-ad-http-crls`"

Then again, RFC4387 predates RFC5280 by ~27 months yet `.6` and `.7` are absent from / not mentioned in RFC5280...

Comment (reaperhulk) [reply docs/x509/reference.rst:3016]: RFC 5280 doesn't obsolete 4387, but it's still odd. I left this in accidentally though -- I'll just expose the one OID for now (we can wait another 5 years for someone to ask for these two).

Comment (jlgaddis) [reply docs/x509/reference.rst]: Right, RFC5280 doesn't mention or reference RFC4387 at all (leaving some unanswered questions, in my opinion).

I, myself, am quite happy with just "`id-ad-caRepository`". If I end up needing the `.48.6` or `.48.7` OIDs later, I'll send a PR then.

Thanks again!

**Pull request merged**